src/lib/y2storage/disk_size.rb

```diff
@@ -549,6 +549,25 @@ def zero?
       @size == 0
     end
 
+    # Test if {DiskSize} is power of 2 (amount of bytes is power of 2).
+    #
+    # @note To check power of two, the amount of bytes is converted to
+    #   a binary number and then the number of 1's is counted.
+    #   A binary number is power of 2 if and only if it contains only an 1.
+    #
+    # @return [Boolean]
+    #
+    # @example
+    #   x = DiskSize.KiB(4) #=> <DiskSize 4.00 KiB (4096)>
+    #   x.power_of_two?     #=> true
+    #
+    #   x = DiskSize.KB(4) #=> <DiskSize 3.91 KiB (4000)>
+    #   x.power_of_two?    #=> false
+    def power_of_two?
+      return false if zero? || unlimited?
+      size.to_s(2).count("1") == 1
```

Comment:
I still haven't decided whether this is very neat or just hacky :wink: (the line between those two is always thin).

Why not something more generic and less surprising like this?
```ruby
def power_of?(exp)
  return false if unlimited?
  (Math.log(size, exp) % 1).zero?
end
```

Reply:
Totally agree (but you cannot say that the hack was quite cute)

Reply:
Yes. It was cool. Unnecessary but cool :-)


![](http://www.just3ds.com/wp-content/uploads/2015/03/Unnecessary-but-cool-invention-just3ds.com_.jpg)